pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -458,7 +458,13 @@ func (s *Reconciler) Delete(ctx context.Context) error {
 	if s.scope.MachineConfig.PublicIP {
 		publicIPName, err := azure.GenerateMachinePublicIPName(s.scope.Cluster.Name, s.scope.Machine.Name)
 		if err != nil {
-			return errors.Wrap(err, "unable to create Public IP")
+			// Only when the generated name is longer than allowed by the Azure portal
```

Comment:
Can we prevent us ever getting into this state by failing in create`?

Reply:
We can't. Anyone can set publicIP field to true aftet an instance was successfully created. In which case err will be non-nil if the machine name is too long.